### PR TITLE
Add runbook to restore stackgres backup

### LINKED ISSUE #9577: Restore Citus Backup
### Problem

Need to be able to restore a backup taken with stackgres

### Solution

- Create runbook for approach to use
- Create any tickets upstream in stackgres or zfs local pv to enable automatic restore without the need for 2x the storage

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds a script to restore stackgres backup

- Add a script to restore citus from a stackgres sharded backup with ZFS snapshots
- Add a simple readme

**Related issue(s)**:

Fixes #9577 

**Notes for reviewer**:

Tested the following scenario:

- simple case, restore the most recent SGShardedBackup
- restore an older SGShardedBackup, any backups after the restore target are removed by the script
- restore an SGShardedBackup in which the coordinator data is backed up from `coord-1` instead of the default coordinator primary `coord-0`

Overall rolling back a ZFS volume to a snapshot and recover the saved WAL files are fast, however it make take longer if the data diff size is larger between the snapshot and the current volume.

The more time consuming part is re-initializing the coordinator replica from the primary, it took about 5 minutes for testnet with 5GB coordinator data. 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -502,82 +502,82 @@ recursively merged into) the current configuration:
...
-| `hedera.mirror.rest.stateproof.streams.cloudProvider`             | S3                      | The cloud provider to download files from. Either `S3` or `GCP`                                                                                                                               |
-| `hedera.mirror.rest.stateproof.streams.endpointOverride`          |                         | Can be specified to download streams from a source other than S3 and GCP. Should be S3 compatible                                                                                             |
-| `hedera.mirror.rest.stateproof.streams.gcpProjectId`              |                         | GCP project id to bill for requests to GCS bucket which has Requester Pays enabled.                                                                                                           |
+| Name                                                               | Default                 | Description                                                                                                                                                                                   |
```
Comment (xin-hedera) [inline docs/configuration.md:505]: just spotless format changes

File: docs/runbook/scripts/utils.sh [hunk 2]
```diff
@@ -16,12 +18,15 @@ function checkCitusMetadataSyncStatus() {
...
-  log "Citus metadata did not sync in 300 seconds"
+  log "Citus metadata did not sync in ${TIMEOUT_SECONDS} seconds"
   exit 1
 }
 
 function doContinue() {
-  read -p "Continue? (Y/N): " confirm && [[ $confirm == [yY] || $confirm == [yY][eE][sS] ]] || exit 1
+  while true; do
+    read -p "Continue? (Y/N): " confirm && [[ $confirm == [yY] || $confirm == [yY][eE][sS] ]] && return || \
+      { [[ -n "$confirm" ]] && exit 1 || true; }
```
Comment (xin-hedera) [inline docs/runbook/scripts/utils.sh:28]: change the function to ignore empty input

File: docs/runbook/scripts/restore-stackgres-backup.sh [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+  done
+
+  log "Waiting for all StackGresCluster pods to be ready"
+  for sts in $(kubectl get sts -l 'app=StackGresCluster' -o name); do
+    expected=$(kubectl get "${sts}" -o jsonpath='{.spec.replicas}')
+    kubectl wait --for=jsonpath='{.status.readyReplicas}'=${expected} "${sts}" --timeout=-1s
+  done
+}
+
+CURRENT_NAMESPACE=$(kubectl config view --minify --output 'jsonpath={..namespace}')
```
Comment (jnels124) [inline docs/runbook/scripts/restore-stackgres-backup.sh:384]: Maybe add to requirements in the `.md`?

File: hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/UsageBasedFeeCalculatorTest.java [hunk 4]
```diff
@@ -285,8 +285,7 @@ void failsWithNseeGetTxnUsageEstimator() {
     void failsWithNseeGetQueryUsageEstimator() {
         FeeData usgPrices = currentPrices.get(DEFAULT);
         final Exception exception = assertThrows(
-                NoSuchElementException.class,
-                () -> subject.estimatePayment(query, usgPrices, at, ANSWER_ONLY));
+                NoSuchElementException.class, () -> subject.estimatePayment(query, usgPrices, at, ANSWER_ONLY));
```
Comment (steven-sheehy) [inline hedera-mirror-web3/src/test/java/com/hedera/services/fees/calculation/UsageBasedFeeCalculatorTest.java:288]: This was fixed on main so might cause merge conflict.

File: docs/configuration.md (hunk 1 above)
Comment (steven-sheehy) [reply docs/configuration.md:505]: Fixed on main already

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: docs/configuration.md (hunk 1 above)
Comment (xin-hedera) [reply docs/configuration.md]: merged again, should be fine now

**Review by steven-sheehy — APPROVED** [review]

**Review by edwin-greene — APPROVED** [review]
Looks good

**Review by jnels124 — APPROVED** [review]
LGTM

**Pull request merged**